conf/modules.config

```diff
@@ -244,13 +265,65 @@ process {
         ]
     }
 
-    withName: SAMTOOLS_FLAGSTAT {
+    withName: ".*MAP:SAMTOOLS_FLAGSTAT" {
+        ext.prefix = { "${meta.id}_${meta.library_id}" }
         publishDir = [
             path: { "${params.outdir}/mapping/" },
             mode: params.publish_dir_mode,
             pattern: '*.flagstat'
         ]
     }
 
-}
+    withName: ".*DEDUPLICATE:PICARD_MARKDUPLICATES" {
+        ext.prefix = { "${meta.id}_${meta.library_id}_${meta.genomic_region}.MarkDuplicates" }
+        ext.args = [
+            "--REMOVE_DUPLICATES",
+            "--VALIDATION_STRINGENCY SILENT",
+            "--ASSUME_SORTED"
+        ].join(' ').trim()
+        publishDir = [
+            enabled: false
+        ]
+    }
+
+    withName: ".*DEDUPLICATE:DEDUP" {
+        ext.args = { params.dedup_all_merged ? '-m' : '' }
```

Comment:
I want to be stricter with the user now with dedup, by only allowing merged-only reads when using DeDup,  at a pipeline level, and do not allow them to use `-m` hacks.

So I would remove this `args`, remove the `arams.dedup_all_merged`, and we should add an error in `eager.nf` saying if `!params.dedupper dedup && !params.merged_only` then 'can't do this!'

This should be added  to the changelog too under  [dev]

Reply:
a) This requires referencing parameter `merged_only` which does not exist at the moment (I guess it will be added with samtools_filter?)
b) Since we are not implementing `AdapterRemovalFixPrefix`, the `-m` option will need to always be specified to `Dedup`

Reply:
For now, have added the suggested error check, but left it commented. you will need to pull the changes from this branch (when done) into the samtools filter PR